packages/platform/atoms/hooks/event-types/useEventTypesPublic.ts

```diff
@@ -0,0 +1,26 @@
+import { useQuery } from "@tanstack/react-query";
+
+import { BASE_URL, API_VERSION, V2_ENDPOINTS, SUCCESS_STATUS } from "@calcom/platform-constants";
+import type { EventTypesPublic } from "@calcom/platform-libraries";
+import type { ApiResponse } from "@calcom/platform-types";
+
+import http from "../../lib/http";
+
+export const QUERY_KEY = "get-private-events";
+export const useEventTypesPublic = (username: string) => {
+  const endpoint = new URL(BASE_URL);
+
+  endpoint.pathname = `api/${API_VERSION}/${V2_ENDPOINTS.eventTypes}/${username}/public`;
+
+  return useQuery({
+    queryKey: [QUERY_KEY],
```

Comment:
`queryKey: [QUERY_KEY, username]`

Reply:
react-query needs to be aware of the username to know that when the props change it should fetch

Reply:
Resolved in [refactor: track dynamic prop in react query](https://github.com/calcom/cal.com/pull/14192/commits/a3d534587f0c354af73a6149356836807f853b98)